test/feature-benchmark/scenarios.py

```diff
@@ -670,6 +670,10 @@ def init(self) -> Action:
 # Make sure we are fully caught up before continuing
 > SELECT COUNT(*) FROM s1;
 {self.n()}
+
+# Give time for any background tasks (e.g. compaction) to settle down
```

Comment:
hmmm im really suspicious of this as the data should all be in trace by the time the `select count(*)` returns the correct count. let me think about this a little bit more

Reply:
we're still working through which of the approaches we like for hooking AllowCompaction up to kafka sources, so it's definitely not trace compaction :)

it could be draining unsealed though. we'll happily forward on a seal to the listener once the seal has been made durable but before the data has been drained into trace. this listener forward is all that is necessary to unblock the dataflow and allow it to serve the SELECT.

it would be straightforward enough to hack a PR that moves the listener forward to be triggered by drain_unsealed instead of when seal is durable. unless there's something else going on (certainly possible), that would be a more principled version of exactly what this PR does. if the hack seems to fix the stability of this benchmark I'd be happy for us to either commit to the move or make it an option hooked up to a cli flag

Reply:
oh i see - are you saying that in the bad state we end up with all of the data in unsealed, not in trace, and then we end up restarting, retracting everything in unsealed, and redoing a bunch of work?. Another way to trigger the bad case more often then would be to call `drain_unsealed` less frequently?